torch_frame/data/multi_embedding_tensor.py

```diff
@@ -117,6 +94,70 @@ def from_tensor_list(
         offset = torch.LongTensor(offset_list)
         return cls(num_rows, num_cols, values, offset)
 
+    def __getitem__(
+        self,
+        index: Any,
+    ) -> Union["MultiEmbeddingTensor", Tensor]:
+        if isinstance(index, tuple) and len(index) == 2 and isinstance(
+                index[0], int) and isinstance(index[1], int):
+            i = self._normalize_index(index[0], dim=0)
+            j = self._normalize_index(index[1], dim=1)
+            return self.values[i, self.offset[j]:self.offset[j + 1]]
+        if isinstance(index, int):
+            index = self._normalize_index(index, dim=0)
+            return MultiEmbeddingTensor(
+                num_rows=1,
+                num_cols=self.num_cols,
+                values=self.values[index].view(1, -1),
+                offset=self.offset,
+            )
+        if isinstance(index, Tensor) and index.ndim == 1:
+            return self.index_select(index, dim=0)
+        if isinstance(index, list):
+            return self.index_select(
+                torch.tensor(index, device=self.device),
+                dim=0,
+            )
+        # TODO(akihironitta): Support more index types
+        raise NotImplementedError
+
+    def index_select(
+        self,
+        index: Tensor,
+        dim: int,
+    ) -> 'MultiEmbeddingTensor':
+        """Returns a :class:`MultiEmbeddingTensor` which indexes the input
+        :class:`MultiEmbeddingTensor` along the specified dimension.
+
+        Args:
+            index (Tensor): A 1-D tensor of indices to select.
+            dim (int): The dimension to index in.
+
+        Returns:
+            MultiEmbeddingTensor: A :class:`MultiEmbeddingTensor` instance.
+        """
+        dim = self._normalize_dim(dim)
+        index = self._normalize_index(index, dim=dim)
+        if dim == 0:
+            return self._row_index_select(index)
+        elif dim == 1:
+            return self._col_index_select(index)
+
+    def _row_index_select(self, index: Tensor) -> 'MultiEmbeddingTensor':
+        if index.numel() == 0:
+            return MultiEmbeddingTensor(
+                num_rows=0,
+                num_cols=self.num_cols,
+                values=torch.tensor([], device=self.device),
+                offset=torch.zeros(1, dtype=torch.long, device=self.device),
```

Comment:
the length of offset should be num_rows (0) * num_cols + 1 = 1.

Reply:
Nvm, I was wrong :( Since we are implementing `MultiEmbeddingTensor` (not nested tensor), we only need to store `num_cols + 1` offset.

Reply:
I don't think we need special handling of `index.numel() == 0` case here.

Reply:
Done! Please have another look

Reply:
Excellent